src/main/java/frc/robot/subsystems/NewVision/NewVisionIO.java

```diff
@@ -0,0 +1,137 @@
+        }
+
+        public Transform3d getLoggedTransform3d(double[] translation, double[] rotation) {
+            Transform3d transform3d = new Transform3d(
+                new Translation3d(
+                    translation[0], 
+                    translation[1], 
+                    translation[2]),
+                new Rotation3d(
+                    new Quaternion(
+                        rotation[0], 
+                        rotation[1], 
+                        rotation[2], 
+                        rotation[3]
+                    )
+                )
+            );
+            return transform3d;
+        }
+
+        public void getLoggedPhotonTrackedTarget(PhotonTrackedTarget target, LogTable table, String name) {
+            double[] translation = table.getDoubleArray("translation " + name, new double[3]);
+            double[] rotation = table.getDoubleArray("rotation " + name, new double[4]);
+            double[] altTranslation = table.getDoubleArray("translation alt " + name, new double[3]);
+            double[] altRotation = table.getDoubleArray("rotation alt " + name, new double[4]);
+
+            List<TargetCorner> detectedCorners = new ArrayList<>();
+            List<TargetCorner> minAreaRectCorners = new ArrayList<>();
+
+            for (int i = 0; i <4; i++) {
+                detectedCorners.add(new TargetCorner(detectedCornersX[i], detectedCornersY[i]));
+                minAreaRectCorners.add(new TargetCorner(minAreaRectCornersX[i], minAreaRectCornersY[i]));
+            }
+            Transform3d pose = getLoggedTransform3d(translation, rotation);
+            Transform3d altPose = getLoggedTransform3d(altTranslation, altRotation);
+            targets.add(
+                new PhotonTrackedTarget(
+                    table.getDouble("yaw " + name, target.getYaw()),
+                    table.getDouble("pitch " + name, target.getPitch()),
+                    table.getDouble("area " + name, target.getArea()),
+                    table.getDouble("skew " + name, target.getSkew()),
+                    (int)(table.getInteger("fiducial id " + name, target.getFiducialId())),
+                    pose,
+                    altPose,
+                    table.getDouble("pose ambiguity " + name, target.getPoseAmbiguity()),
+                    minAreaRectCorners,
+                    detectedCorners)
+            );
+        }
+
+        @Override
+        public void toLog(LogTable table) {
+            table.put("timestamp", timestamp);
+            table.put("latency", timeSinceLastTimestamp);
+            for (int i = 0; i < targets.size(); i++) {
+                logPhotonTrackedTarget(targets.get(i), table, String.valueOf(i));
+            }
+        }
+        @Override
+        public void fromLog(LogTable table) {
+            timestamp = table.getDouble("timestamp", timestamp);
+            timeSinceLastTimestamp = table.getDouble("latency", timeSinceLastTimestamp);
+            
```

Comment:
need to get all the photon targets in `fromLog()`. might need to log number of targets